test/conformance/README.md

```diff
@@ -0,0 +1,77 @@
+# Conformance tests for sgID SDK
+
+## What is the conformance test suite
+
+OpenID Foundation provides an open-sourced [suite of conformance tests](https://www.certification.openid.net) for relying parties (RPs) and identity providers (IDPs) to verify that their implementation conforms to the standards laid out in the OIDC RFC.
+
+To ensure that our SDKs conform to the OIDC standards, we will use a demo app that utilizes our SDK to act as an RP against the conformance tests.
+
+## How it works
+
+The `demo-rp` directory contains the demo app that will be interacting with the conformance tests.
+
+- This app is based off the Next.js CSR example app under `examples/nextjs-csr`.
+- Additionally, this demo app overrides the endpoints to point to the conformance tests instead of the sgID transaction server (the overriding can be found under `test/conformance/demo-rp/src/lib/sgidClient.ts`).
+
+The `suite` directory contains the scripts that interact with the conformance test server to start the test plan and test modules, and interact with the demo app through API calls.
+
+- `test.py` is the main test runner that defines the config of the conformance test plan and runs each test module
+- `conformance.py` (modified based off this [tutorial](https://gitlab.com/openid/conformance-suite-automated-testing-tutorial)) is a wrapper over the [API calls](https://www.certification.openid.net/api-document.html) to the conformance test server and also includes the logic to interact with the locally hosted demo app
+- `modules.py` specifies the test modules (in the `oidcc-client-test-plan` test plan) to run and their expected final statuses and results
+
+## Setting up the environment for the tests
+
+Before you can run the tests, you will need to:
+
+1. Build the sgID SDK
+   - Run `npm run build` in the root directory (i.e. `/sgid-client`)
+1. Run the demo app locally
+   - Run `cd test/conformance/demo-rp` to move to the `demo-rp` directory
+   - Run `npm i` to install the dependencies
+1. Run the test runner script
+   - Create a new terminal instance
+   - Run `cd test/conformance/suite` to move to the `suite` directory
+   - Run
+     ```
+     python3 -m venv .venv
+     ```
+     to setup a Python virtual env
+   - Run `cat .env.example > .env` and repalce the `CONFORMANCE_TOKEN` with your [conformance testing permanent token](https://www.certification.openid.net/tokens.html) to setup your environment variables
+   - Run `pip install -r requirements.txt` to install the requirements
```

Comment:
have we considered running these tests as part of our CI/CD pipeline? if we've decided against it, might be good to document why, probably in the PR description or in this README

Reply:
thanks for the comment - documented in the README here [aebdb25](https://github.com/opengovsg/sgid-client/pull/60/commits/aebdb25a185fd56e589696002ad511374a8e03b7)